tests/test_json_to_parquet.py

```diff
@@ -13,597 +13,570 @@
-                {
-                    "Size": 1,
-                    "Key": os.path.join(dataset_key_prefix, "object2"),
-                    "Bucket": artifact_bucket
-                }
-            ]
+        "Contents": [
+            {
+                "Size": 0,
+                "Key": os.path.join(dataset_key_prefix, "object0"),
+                "Bucket": artifact_bucket,
+            },
+            {
+                "Size": 1,
+                "Key": os.path.join(dataset_key_prefix, "object1"),
+                "Bucket": artifact_bucket,
+            },
+            {
+                "Size": 1,
+                "Key": os.path.join(dataset_key_prefix, "object2"),
+                "Bucket": artifact_bucket,
+            },
+        ]
     }
     mock_client.list_objects_v2.return_value = objects_to_copy
     result = {
-            "mock_client": mock_client,
-            "objects_to_copy": objects_to_copy,
-            "workflow_run_datetime": workflow_run_datetime
+        "mock_client": mock_client,
+        "objects_to_copy": objects_to_copy,
+        "workflow_run_datetime": workflow_run_datetime,
     }
     return result
 
+
 @pytest.fixture()
 def spark_session():
     spark_session = SparkSession.builder.getOrCreate()
     return spark_session
 
+
 @pytest.fixture()
 def sample_table_data():
     sample_table_data = {
-            "name": ["John", "John", "Jane", "Bob"],
-            "age": [25, 25, 30, 22],
-            "city": ["New York", "Chicago", "San Francisco", "Los Angeles"],
-            "GlobalKey": ["1", "1", "2", "3"],
-            "export_end_date": [
-                "2023-05-12T00:00:00",
-                "2023-05-13T00:00:00",
-                "2023-05-13T00:00:00",
-                "2023-05-14T00:00:00"
-            ]
+        "name": ["John", "John", "Jane", "Bob"],
+        "age": [25, 25, 30, 22],
+        "city": ["New York", "Chicago", "San Francisco", "Los Angeles"],
+        "GlobalKey": ["1", "1", "2", "3"],
+        "export_end_date": [
+            "2023-05-12T00:00:00",
+            "2023-05-13T00:00:00",
+            "2023-05-13T00:00:00",
+            "2023-05-14T00:00:00",
+        ],
     }
     return sample_table_data
 
+
 @pytest.fixture()
 def sample_table_inserted_date_data():
     sample_table_data = {
-            "name": ["John", "John", "Jane", "Bob"],
-            "age": [25, 25, 30, 22],
-            "city": ["New York", "Chicago", "San Francisco", "Los Angeles"],
-            "GlobalKey": ["1", "1", "2", "3"],
-            "InsertedDate": [
-                "2023-05-12T00:00:00",
-                "2023-05-13T00:00:00",
-                "2023-05-13T00:00:00",
-                "2023-05-14T00:00:00"
-            ],
-            "export_end_date": [
-                "2023-05-14T00:00:00",
-                "2023-05-14T00:00:00",
-                "2023-05-14T00:00:00",
-                "2023-05-14T00:00:00"
-            ]
+        "name": ["John", "John", "Jane", "Bob", "Bob_2"],
```

Comment:
I added to this test case data with the problem situation with matching `InsertedDate` and a newer `export_end_date`